catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/XstreamPathConverter.java

```diff
@@ -164,13 +166,11 @@ protected boolean doBasicPathsMatch(final Path pathObj1, final Path pathObj2) {
      */
     private boolean endsMatch(char[] path1, int index1, char[] path2, int index2) {
```

Comment:
Also, I believe a lot the strings being compared are repeated, so maybe caching would be a good option as well.